lib/terraform/runner.rb

```diff
@@ -0,0 +1,293 @@
+        'c158d710-d91c-11ed-9fee-d93323035b4e'
+      end
+
+      # Create TerraformRunner Stack Job
+      def create_stack_job(
+        template_path,
+        input_vars: [],
+        tags: nil,
+        credentials: [],
+        env_vars: {},
+        name: "stack-#{rand(36**8).to_s(36)}"
+      )
+        _log.info("create_stack_job for #{template_path}")
+        tenant_id = stack_tenant_id
+
+        # Temp Zip File
+        # zip_file_path = Tempfile.new(%w/tmp .zip/).path
+        zip_file_path = Tempfile.new(%w[tmp .zip]).path
+        create_zip_file_from_directory(zip_file_path, template_path)
+        zip_file_hash = Base64.encode64(File.binread(zip_file_path))
+
+        # TODO: use tags,env_vars
+        payload = JSON.generate(
+          {
+            :cloud_providers => credentials,
+            :name            => name,
+            :tenantId        => tenant_id,
+            :templateZipFile => zip_file_hash,
+            :parameters      => convert_to_cam_parameters(input_vars)
+          }
+        )
+        # _log.debug("Payload:>\n, #{payload}")
+        http_response = terraform_runner_client['api/stack/create'].post(
+          payload, :content_type => 'application/json'
+        )
+        _log.info("==== http_response.body: \n #{http_response.body}")
+        Terraform::Runner::Response.parsed_response(http_response)
+      ensure
+        # cleanup temp zip file
+        FileUtils.rm_rf(zip_file_path) if zip_file_path
+        _log.info("Deleted #{zip_file_path}")
+      end
+
+      # Retrieve TerraformRunner Stack Job details
+      def retrieve_stack_job(stack_id)
+        payload = JSON.generate(
+          {
+            :stack_id => stack_id
+          }
+        )
+        http_response = terraform_runner_client['api/stack/retrieve'].post(
+          payload, :content_type => 'application/json'
+        )
+        _log.info("==== Retrieve Stack Response: \n #{http_response.body}")
+        Terraform::Runner::Response.parsed_response(http_response)
+      end
+
+      # Cancel/Stop running TerraformRunner Stack Job
+      def cancel_stack_job(stack_id)
+        payload = JSON.generate(
+          {
+            :stack_id => stack_id
+          }
+        )
+        http_response = terraform_runner_client['api/stack/cancel'].post(
+          payload, :content_type => 'application/json'
+        )
+        _log.info("==== Cancel Stack Response: \n #{http_response.body}")
+        Terraform::Runner::Response.parsed_response(http_response)
+      end
+
+      # Wait for TerraformRunner Stack Job to complete
+      def wait_until_completes(stack_id)
+        interval_in_secs = stack_job_interval_in_secs
+        max_time_in_secs = stack_job_max_time_in_secs
+
+        response = nil
+        Timeout.timeout(max_time_in_secs) do
+          _log.info("Starting wait for terraform-runner/stack/#{stack_id} completes ...")
+          i = 0
+          loop do
+            _log.info(i)
```

Comment:
Some of these log lines might need to become debug lines.  We'll need to see what a full runner log looks like.

This one in particular looks like strict debug and not intended to be merged like this.

Reply:
using _log.debug